### ISSUE TITLE
defrag_task will not run when use dbnum=1

### ISSUE THREAD

**Issue by bbqccx** [dragonflydb/dragonfly#2762]
When I use defrag_mem_test.py to test memory defragmentation, I find that when starting Dragonfly with dbnum=1, the defrag_task will not be run even if the defragmentation conditions are met. However, when dbnum=1 is removed or set to dbnum=2, the defrag_task will be run when the defragmentation conditions are met.

dragonfly  --maxmemory 10G --proactor_threads=8 --bind localhost --dbnum=1 --mem_defrag_threshold=0.01
defrag_mem_test.py  -k 6000000 -v 200

**Comment by bbqccx** [comment]
os is debian 11, 5.10.0-26-amd64 #1 SMP Debian 5.10.197-1 (2023-09-29) x86_64 GNU/Linux

**Comment by romange** [comment]
yes, indeed we do not have the support for this with epoll. I suggest using debian 12

**Comment by dranikpg** [comment]
Close by https://github.com/romange/helio/pull/317
